File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_to_index = {}
        seen = set()

        for i, num in enumerate(nums):
            tmp = target - num
            if num in seen:
                return [i, num_to_index[num]]

            num_to_index[tmp] = i
            seen.add(tmp)

File: 0001-two-sum/0001-two-sum.ts
function twoSum(nums: number[], target: number): number[] {
  const tracker: { [key: number]: number } = {}

  for (let i=0; i < nums.length; i++) {
    const num = nums[i]
    
    if (tracker.hasOwnProperty(num)) {
        return [i, tracker[num]]
    }
    tracker[target - num] = i
  }  
  return null;
};